serde-generate/src/golang.rs

```diff
@@ -443,10 +443,10 @@ obj := make(map[{0}]{1})
 previous_slice := serde.Slice {{ 0, 0 }}
 for i := 0; i < length; i++ {{
 	var slice serde.Slice
-	slice.Start = deserializer.GetBufferOffset()
+	slice.Start = uint64(deserializer.GetBufferOffset())
```

Comment:
It looks like the definition of `serde.Slice` is right and `deserializer.GetBufferOffset()` should be an `uint64`. (Once again Java is stuck in a 32-bit world and gave us the wrong impression.)

Reply:
1. I don't think it really matters to support uint64, as 99.99999999999999999% cases will not hit int limit, because it means 2^31-1 bytes, it's 2GB data, and it probably not going to performance well.
2. as I used bytes.Buffer internally, the Buffer#Len is int type, if we changed to uint64, it basically means we need re-impl bytes.Buffer, given the above 1, I don't think it worth the effort.